### PR TITLE
Prosemirror comments

### PR DESCRIPTION
**Pull request by viktor992**
Added prosemirror editor to comment containers and inputs.

### CODE REVIEW

**Comment by mitar** [comment]
I suggest you rebase this on current master?

**Comment by viktor992** [comment]
Ready for review.

File: client/components/utils/link.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      linkDialogRef.openLinkDialog(selectedExistingLinks);
+      return true;
+    }
+    else {
+      return toggleMark(state.schema.marks.link)(state);
+    }
+  };
+}
+
+export function _clearLink(editorView) {
```
Comment (mitar) [inline client/components/utils/link.js]: Why are these private? You moved the from the class, and you are exporting them, so no need for `_` anymore.

File: client/components/comment-editor.vue [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          ],
+        });
+        this.$commentEditorView = new EditorView({mount: this.$refs.commentBody}, {
+          state,
+          dispatchTransaction: (transaction) => {
+            const newState = this.$commentEditorView.state.apply(transaction);
+            this.$commentEditorView.updateState(newState);
+            this.$commentEditorView.state = newState;
+
+            const fragment = DOMSerializer.fromSchema(schema).serializeFragment(newState.doc.content);
```
Comment (mitar) [inline client/components/comment-editor.vue]: What? What? What is this? Why are you generating a HTML snippet, which you then store in the database, and which you then read back, parse it back into ProseMirror, for ProseMirror to render?

I would understand if you would generate HTML snippet (on server side!) and store it to render it directly. Or if you would store directly document state into the database, and then load it on the client to use ProseMirror to render it in read-only mode. But not both.

File: client/components/comment.vue [hunk 3]
```diff
@@ -23,6 +23,7 @@
 </template>
 
 <script>
+
```
Comment (mitar) [inline client/components/comment.vue]: Please do not include empty line at the beginning.

File: client/components/thread.vue [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        this.$emit("hideNewCommentForm");
+        this.$refs.threadInput.clearEditor();
+      },
+    },
+  };
+
+  export default component;
+</script>
+
+<style>
```
Comment (mitar) [inline client/components/thread.vue]: Please use `<style lang="scss">` everywhere.

Comment (mitar) [reply client/components/thread.vue]: Then you can nest selectors.

**Comment by mitar** [comment]
So it is not yet perfect. I will merge this now and then try to improve things a bit.

**Pull request merged**